applications/app/services/InteractivePicker.scala

```diff
@@ -0,0 +1,56 @@
+package services
+
+import com.gu.contentapi.client.model.v1.{CapiDateTime, ItemResponse, Tag}
+import conf.switches.Switches.InteractivePickerFeature
+import play.api.mvc.RequestHeader
+import implicits.Requests._
+import org.joda.time.{DateTime, DateTimeZone, LocalDate}
+import org.joda.time.format.DateTimeFormat
+
+sealed trait RenderingTier
+object DotcomRendering extends RenderingTier
+object FrontendLegacy extends RenderingTier
+object USElectionTracker2020AmpPage extends RenderingTier
+
+case class InteractivePickerInputData(datetime: CapiDateTime, tags: List[Tag])
+
+object InteractivePicker {
+
+  val migratedPaths = List(
+    "/sport/ng-interactive/2018/dec/26/lebron-james-comments-nba-nfl-divide",
+  )
+
+  val tagsBlockList: Set[String] = Set(
+    "tracking/platformfunctional/dcrblacklist",
+  )
+
+  def ensureStartingForwardSlash(str: String): String = {
+    if (!str.startsWith("/")) ("/" + str) else str
+  }
+
+  def dateIsPostTransition(date: String): Boolean = {
+    // It's the responsibility of the caller of this function to ensure that the date
+    // is given as string in format "YYYY-MM-DD", otherwise "results may vary".
+    date >= "2021-06-21"
+  }
+
+  def isInTagBlockList(tags: List[Tag]): Boolean = {
```

Comment:
It's not clear that we need a general tag system here. We only plan to support a single opt-out tag value (as far as I know) so let's target that and name accordingly. E.g.

`def isNotOptOut(tags: List[Tag]): Boolean`

or something. And just inline `"tracking/platformfunctional/dcrblacklist"` into a val in the method to reduce surface area. Did we also rename the tag to 'dcrblocklist' as well?

Reply:
The tag name has been updated, but not the id string

Reply:
Ah okay fair enough.